### ISSUE TITLE
`componentWillReceiveProps`: how to know when dynamic segment changes?

### ISSUE THREAD

**Issue by ruggeri** [rackt/react-router#584]
**tl;dr: what is the recommended way in `componentWillReceiveProps` to find the *previous dynamic segment*?**

Longer:

I understand the note about dynamic segments: the lifecycle hook to fetch data when a dynamic segment changes is `componentWillReceiveProps`. Cool.

In React, `componentWillReceiveProps` receives as an argument the `nextProps`. These can be compared to **this.props** and used to determine whether a relevant property changed and data should be fetched.

What I want to do is **only fetch new data in `componentWillReceiveProps` if the dynamic segment changes**. Significantly, in the presence of other props, I cannot assume that `componentWillReceiveProps` is called only when the dynamic segment changes. **However I haven't found a way to recover the previous routing params**.

I can easily work around by setting an instance variable in `componentWillReceiveProps` to record the current routing params for comparison on the next invocation of `componentWillReceiveProps`. That seems sort of nasty and stateful.

I hope I haven't misunderstood. But if I have understood correctly, then the routing params context is mutable and the previous route is lost, and cannot be compared against. That seems contrary to the concept of `componentWillRecieveProps`.

**Comment by rpflorence** [comment]
see https://github.com/rackt/react-router/issues/579#issuecomment-72350478
